pkg-manager/core/src/install/index.ts

```diff
@@ -394,6 +396,7 @@ export async function mutateModules (
           ctx.wantedLockfile.lockfileVersion === LOCKFILE_VERSION_V6 ||
           ctx.wantedLockfile.lockfileVersion === '6.1'
         ) &&
+        allCatalogsAreUpToDate(opts.catalogs, ctx.wantedLockfile.catalogs) &&
         await allProjectsAreUpToDate(Object.values(ctx.projects), {
```

Comment:
can't these be combined? `allProjectsAreUpToDate` is there for checking if the lockfile is up-to-date. It should probably check catalogs as well.

Reply:
Ah I see. Yup, I can do that. The `allProjectsAreUpToDate` was doing something a bit different right now, but I didn't realize the intention was to check of the lockfile is up to date in general. I thought it was just for the `importers` section.

Reply:
good point... I think it is fine to extend it. Or create a new function and call both allCatalogsAreUpToDate and allProjectsAreUpToDate from it.

Reply:
Done! I ended up adding it to `allProjectsAreUpToDate` with a comment about how the `catalogs` block affects projects.